drivers/accel/adxl314/adxl314.c

```diff
@@ -0,0 +1,1119 @@
+					      DISABLE_E, ADXL314_POWER_CTL_MEASURE);
+		break;
+	case ADXL314_MEAS:
+		ret = adxl314_control_reg_bit(dev, ADXL314_REG_POWER_CTL,
+					      ENABLE_E, ADXL314_POWER_CTL_MEASURE);
+		break;
+	default:
+		pr_err("ADXL314 operation mode not supported!\n");
+		return -ENOSYS;
+	}
+
+	if (!ret)
+		dev->op_mode = op_mode;
+
+	return ret;
+}
+
+/*******************************************************************************
+ * @brief Get the current operation mode of the device.
+ *
+ * @param dev     - The device structure.
+ * @param op_mode - Read operation mode.
+ *
+ * @return ret    - Result of the operation procedure (0 - success, negative
+ *                  value - failure).
+*******************************************************************************/
+int adxl314_get_op_mode(struct adxl314_dev *dev,
+			enum adxl314_op_mode *op_mode)
+{
+	int ret;
+	uint8_t reg_value = 0x00;
+
+	ret = adxl314_read_device_data(dev, ADXL314_REG_POWER_CTL, 1, &reg_value);
+	if (ret)
+		return ret;
+
+	if (reg_value & ADXL314_POWER_CTL_MEASURE)
+		*op_mode = ADXL314_MEAS;
+	else
+		*op_mode = ADXL314_STDBY;
+
+	return ret;
+}
+
+/*******************************************************************************
+ * @brief Set offset for each axis. Scale factor of 1.56 g/LSB.
+ *
+ * @param dev      - The device structure.
+ * @param offset   - Offset.
+ * @param axis     - Axis to apply offset.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int adxl314_set_offset(struct adxl314_dev *dev, uint8_t offset,
+		       enum adxl314_axis axis)
+{
+	int ret;
+
+	if (axis > ADXL314_Z_AXIS)
+		return -EINVAL;
+
+	/* Write data to register for setting odr. */
+	ret = adxl314_write_device_data(dev, ADXL314_REG_OFS_AXIS(axis), 1, &offset);
+	if (ret)
+		return ret;
+
+	switch (axis) {
+	case ADXL314_X_AXIS:
+		dev->x_offset = offset;
+		break;
+	case ADXL314_Y_AXIS:
+		dev->y_offset = offset;
+		break;
+	case ADXL314_Z_AXIS:
+		dev->z_offset = offset;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return ret;
+}
+
+/*******************************************************************************
+ * @brief Read the 3-axis raw data from the accelerometer.
+ *
+ * @param dev - The device structure.
+ * @param x   - X-axis data (as two's complement).
+ * @param y   - Y-axis data (as two's complement).
+ * @param z   - Z-axis data (as two's complement).
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int adxl314_get_raw_xyz(struct adxl314_dev *dev,
+			int16_t *x,
+			int16_t *y,
+			int16_t *z)
+{
+	int ret;
+	uint8_t xyz_values[6] = {0};
+
+	ret = adxl314_read_device_data(dev,
+				       ADXL314_REG_DATA_AXIS(0),
+				       ADXL314_REGS_PER_ENTRY,
+				       xyz_values);
+	if (ret)
+		return ret;
+
+	/* Result is 13 bits long with sign extended. */
```

Comment:
is it 13-bit or 16-bit?

Reply:
It is a 13-bit result with sign extension.